hosted/helpers/helper_common.go

```diff
@@ -354,3 +337,27 @@ func DefaultK8sVersion(descVersions []string, forUpgrade bool) (string, error) {
 	}
 	return descVersions[1], nil
 }
+
+func CreateCloudCredentials(client *rancher.Client) (*v1.SteveAPIObject, error) {
+	var (
+		err                   error
+		cloudCredential       *v1.SteveAPIObject
+		cloudCredentialConfig cloudcredentials.CloudCredential
+	)
+
+	switch Provider {
+	case "aks":
+		cloudCredentialConfig = cloudcredentials.LoadCloudCredential("azure")
+		cloudCredential, err = azure.CreateAzureCloudCredentials(client, cloudCredentialConfig)
+		Expect(err).To(BeNil())
+	case "eks":
+		cloudCredentialConfig = cloudcredentials.LoadCloudCredential("aws")
+		cloudCredential, err = aws.CreateAWSCloudCredentials(client, cloudCredentialConfig)
+		Expect(err).To(BeNil())
+	case "gke":
+		cloudCredentialConfig = cloudcredentials.LoadCloudCredential("google")
+		cloudCredential, err = google.CreateGoogleCloudCredentials(client, cloudCredentialConfig)
+		Expect(err).To(BeNil())
+	}
+	return cloudCredential, nil
```

Comment:
I do not see cloudCredential *v1.SteveAPIObject is being used anywhere else except to determine cloudCredentialID, can we can remove it and simply return the ID from this func?
```suggestion
	return fmt.Sprintf("%s:%s", cloudCredential.Namespace, cloudCredential.Name), nil
```

Reply:
Done, tested locally and working fine.